src/query/query_executor.c

```diff
@@ -7916,6 +7957,13 @@ qexec_intprt_fnc (THREAD_ENTRY * thread_p, XASL_NODE * xasl, XASL_STATE * xasl_s
 	  qexec_clear_all_lists (thread_p, xasl);
 	}
 
+      if (max_recursive_iterations_reached)
+	{
+	  xb_scan = S_END;
+	  er_set (ER_NOTIFICATION_SEVERITY, ARG_FILE_LINE, ER_CTE_MAX_RECURSION_REACHED, 1, xasl->max_iterations);
+	  break;
```

Comment:
Please confirm other's behavior. SQLServer raises an error for the case. 

I think return `S_ERROR` and raise an error here is clearer. Return a partial result may confuse users.